internal/transport/transport_test.go

```diff
@@ -179,3 +182,57 @@ func TestTimeseries(t *testing.T) {
 		}
 	})
 }
+
+func TestNewFetchConfig(t *testing.T) {
+	t.Parallel()
+
+	testURL, err := url.Parse("https//api.test.com/")
```

Comment:
Let's avoid using URL's with a TLD, instead do something like "https://fuzz".
```suggestion
	testURL, err := url.Parse("https://fuzz")
```